extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -899,103 +876,162 @@ export class AmpLightboxGallery extends AMP.BaseElement {
-            opacity: numeric(1, 0.01),
-          }), 0.1, EXIT_CURVE_);
-
-          return this.measureMutateElement(
-              () => {
-                const rect = layoutRectFromDomRect(sourceElement
-                    ./*OK*/getBoundingClientRect());
-                setStyles(clone, {
-                  position: 'absolute',
-                  top: px(rect.top),
-                  left: px(rect.left),
-                  width: px(rect.width),
-                  height: px(rect.height),
-                  transformOrigin: 'top left',
-                  willChange: 'transform',
-                });
-                const dx = imageBox.left - rect.left;
-                const dy = imageBox.top - rect.top;
-                const scaleX = rect.width != 0 ?
-                  imageBox.width / rect.width : 1;
-                const viewportHeight = this.getViewport().getSize().height;
-                duration = this.getTransitionDuration_(Math.abs(dy),
-                    viewportHeight);
-
-                // Animate the position and scale of the transition image to its
-                // final lightbox destination in the middle of the page
-                anim.add(0, setStylesTransition(clone, {
-                  transform: concatTransition([
-                    translate(numeric(0, dx), numeric(0, dy)),
-                    scale(numeric(1, scaleX)),
-                  ]),
-                }), MOTION_DURATION_RATIO, ENTER_CURVE_);
-              },
-              () => {
-                const carousel = dev().assertElement(this.carousel_);
-                toggle(carousel, true);
-                setStyles(carousel, {
-                  opacity: 0,
-                });
-                sourceElement.classList.add('i-amphtml-ghost');
-                this.getAmpDoc().getBody().appendChild(transLayer);
-              });
-        }).then(() => {
-          return anim.start(duration).thenAlways(() => {
-            return this.mutateElement(() => {
-              setStyles(this.element, {opacity: ''});
-              setStyles(dev().assertElement(this.carousel_), {opacity: ''});
-              sourceElement.classList.remove('i-amphtml-ghost');
-              if (transLayer) {
-                this.getAmpDoc().getBody().removeChild(transLayer);
-              }
-            });
-          });
+          const lightboxImg = imageViewer.getImage();
+          const sourceImg = childElementByTag(sourceElement, 'img');
+
+          return this.runImgTransition_(
+              enter ? sourceImg : lightboxImg,
+              enter ? lightboxImg : sourceImg,
+              enter);
         });
   }
 
+  /**
+   * Gets the duration for how long to animate the transition, based on the
+   * distance and size of the viewport.
+   * @param {!Element} srcImg The img to transition from.
+   * @param {!Element} targetImg The img to transition to.
+   * @return {number} How long to animate for, in milliseconds.
```

Comment:
Do we still need the `@private` annotation? I notice that it's been left out in a few of your new functions.

Reply:
Done.